homeassistant/components/anova_sous_vide/__init__.py

```diff
@@ -0,0 +1,40 @@
+"""The Anova integration."""
+from __future__ import annotations
+
+from anova_wifi import AnovaOffline, AnovaPrecisionCooker, AnovaPrecisionCookerSensor
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import ANOVA_CLIENT, ANOVA_FIRMWARE_VERSION, DOMAIN
+
+PLATFORMS = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Anova from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+    apc = AnovaPrecisionCooker()
+    try:
+        update = await apc.update(entry.data["device_id"])
+    except AnovaOffline as ex:
+        raise ConfigEntryNotReady("Can not connect to the sous vide") from ex
+    hass.data[DOMAIN][entry.entry_id] = {
+        ANOVA_CLIENT: AnovaPrecisionCooker(),
+        ANOVA_FIRMWARE_VERSION: update["sensors"][
+            AnovaPrecisionCookerSensor.FIRMWARE_VERSION
+        ],
+    }
```

Comment:
It would better to store a dataclass here (see lookin/model*.py for an example)

Reply:
I realized that both of those pieces of data are just used in the coordinator init. so instead I set hass.data[domain][entry.entry_id] = the coordinator. Is that okay?

Reply:
Yes thats good as well